### PR TITLE
Add RevisionID to APISpec

### LINKED ISSUE #357: Feature request: Add support for revision ID for APISpec
You have an idea? We'd love to hear it!

## What's missing?
When reporting errors related to an APISpec, it would be helpful to know the revision of the API Spec that generated the error, in case the two are correlated.

## Your preferred solution
Add a field in APISpec to reflect its revision ID.

### PR DESCRIPTION
**Pull request by emschwar**
* In order to avoid passing yet more arguments into apigeeDynamicDataHeaders(),
  replace the `basepath` arg with the APISpec as a whole, so that we can pull
  whatever other data we need out of it as we go on.

Closes #357

### CODE REVIEW

File: config/env_spec.go [hunk 1]
```diff
@@ -225,6 +225,8 @@ type APISpec struct {
 	// ID of the API, used to match the api_source of API Product Operations.
 	ID string `yaml:"id" mapstructure:"id"`
 
+	RevisionID string `yaml:"revision_id" mapstructure:"revision_id,omitempty"`
```
Comment (rockspore) [inline config/env_spec.go:228]: Please add comments with `// RevisionID ...`

Comment (rockspore) [reply config/env_spec.go:228]: Also remove the omitempty in the mapstructure tag.

**Comment by rockspore** [comment]
The test is also failing, please take a look.

File: config/env_spec.go (hunk 1 above)
Comment (emschwar) [reply config/env_spec.go:229]: Done.

**Comment by emschwar** [comment]
> The test is also failing, please take a look.

I got it passing-- maybe not the best way; let me know if it needs a fix.

File: config/env_spec.go (hunk 1 above)
Comment (rockspore) [reply config/env_spec.go:229]: nit: please remove the omitempty part

**Review by rockspore — APPROVED** [review]

File: config/env_spec.go (hunk 1 above)
Comment (emschwar) [reply config/env_spec.go]: Done.

**Pull request merged**